### DISCUSSION TITLE
Use DnD when tree is created by HTML (with ASP.NET ViewModel)?

### DISCUSSION THREAD

**Discussion by MarcoLLeite** [vakata/jstree#2692] [Q&A]
Hi guys, 

I have a tree that is created by HTML (ASP.NET View Models).
But I cannot activate the "dnd ", I added the plugin but it won't work. I get the red cross every time I try to grab something and move it.

Here's my js to create the three:
![image](https://user-images.githubusercontent.com/3867346/215560148-bbe49d74-d2c3-490c-8d3b-3806be63ccc6.png)

My HTML on the main page:
![image](https://user-images.githubusercontent.com/3867346/215560232-07b5b2bc-a597-4df7-a124-a5b8c92d0fcd.png)
    
  

And my partial view code ("LocationTreeNode):
![image](https://user-images.githubusercontent.com/3867346/215560287-4db62c51-903f-43f7-a0e6-17fe31742fee.png)

*The type is dynamic, but I set it to "building" in the code above to simplify things.

Can somebody help? Is this supported?

**Comment by vakata** [comment]
You need to set check_callback (to true for example).
https://www.jstree.com/api/#/?q=check_callba&f=$.jstree.defaults.core.check_callback
